TCAT/Cells/RouteTableViewCell.swift

```diff
@@ -145,110 +184,131 @@ class RouteTableViewCell: UITableViewCell {
-        }
-        
-    }
-
-    private func setLiveElements(withStartTime startTime: Date?, withDelay delay: Int?) {
-        if let delay = delay, let startTime = startTime {
-            let oneMinute = 60
-            if delay >= oneMinute {
-                liveLabel.textColor = .liveRedColor
-                liveLabel.text = "Late - \(Time.timeString(from: startTime.addingTimeInterval(TimeInterval(delay))))"
-                liveIndicatorView.setColor(to: .liveRedColor)
-                
-                let time = Time.timeString(from: Date(), to: startTime.addingTimeInterval(TimeInterval(delay)))
-                departureTimeLabel.text = "Board in \(time)"
-                departureTimeLabel.sizeToFit()
-                departureTimeLabel.textColor = .liveRedColor
-            } else {
-                liveLabel.textColor = .liveGreenColor
-                liveLabel.text = "On Time"
-                liveIndicatorView.setColor(to:.liveGreenColor)
-                
-                departureTimeLabel.textColor = .liveGreenColor
-            }
-            liveLabel.sizeToFit()
-            positionLiveIndicatorView(usingLiveLabel: liveLabel)
-        } else {
-            liveIndicatorView.setColor(to: .clear)
-            liveLabel.textColor = .clear
+            self.hideLiveElements()
             if let route = route {
-                setDepartureTime(withDepartureTime: route.departureTime, withWalkingRoute: route.isRawWalkingRoute(), withSearchTime: searchTime, withSearchTimeType: searchTimeType)
+                self.setDepartureTimeAndLiveElements(withRoute: route, withSearchTime: self.searchTime, withSearchTimeType: self.searchTimeType)
             }
         }
     }
 
-
+    private func hideLiveElements() {
+        liveIndicatorView.setColor(to: .clear)
+        liveLabel.textColor = .clear
+    }
+    
+    private func setLiveElementsOnTime() {
+        liveLabel.textColor = .liveGreenColor
+        liveLabel.text = "On Time"
+        liveIndicatorView.setColor(to:.liveGreenColor)
+        
+        liveLabel.sizeToFit()
+        positionLiveIndicatorView(usingLiveLabel: liveLabel)
+    }
+    
+    private func setDepartureTimeOnTime() {
+        departureTimeLabel.textColor = .liveGreenColor
+        arrowImageView.tintColor = .primaryTextColor
+    }
+    
+    private func setLiveElementsToLate(withDepartureTime departureTime: Date) {
+        liveLabel.textColor = .liveRedColor
+        liveLabel.text = "Late - \(Time.timeString(from: departureTime))"
+        liveIndicatorView.setColor(to: .liveRedColor)
+        
+        liveLabel.sizeToFit()
+        positionLiveIndicatorView(usingLiveLabel: liveLabel)
+    }
+    
+    private func setDepartureTime(withStartTime startTime: Date, withDepartureTime departureTime: Date, late: Bool) {
+        let boardTime = Time.timeString(from: startTime, to: departureTime)
+        departureTimeLabel.text = boardTime == "0 min" ? "Board now" : "Board in \(boardTime)"
```

Comment:
Should we discuss maybe some better language for this, since it's confusing?

Reply:
yes wanted to talk about board in with yall